src/main/java/com/billingapi/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,147 @@
+import javax.validation.ConstraintViolationException;
+
+
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+
+    /*Exception Handling for InvoiceNotFoundException
+    Using InvoiceNotFoundException ,to check invoice is find or Not for given invoiceId
+    * @param ex InvoiceNotFoundException exception to handle
+	 * @param request the current request
+	 * @return a  ResponseEntity for the response to use
+     */
+
+    @ExceptionHandler(InvoiceNotFoundException.class)
+    public ResponseEntity<ExceptionResponse> invoiceNotFoundException(Exception ex, WebRequest request) {
+        ExceptionResponse exceptionResponse =
+                new ExceptionResponse(ex.getMessage());
+        return new ResponseEntity<>(exceptionResponse, HttpStatus.NOT_FOUND);
+
+    }
+
+    /*
+    Exception Handling of   handleHttpRequestMethodNotSupported
+
+    Using HttpRequestMethodNotSupportedException check Method is supported or not
+    * @param ex HttpRequestMethodNotSupportedException exception to handle
+	 * @param request the current request
+	 * @return a  ResponseEntity for the response to use
+     */
+    @ExceptionHandler(HttpRequestMethodNotSupportedException.class)
+    public ResponseEntity<Object> handleHttpRequestMethodNotSupported(
+            HttpRequestMethodNotSupportedException ex, WebRequest request) {
+        ExceptionResponse exceptionResponse =
+                new ExceptionResponse(ex.getMessage());
+        return new ResponseEntity<Object>(exceptionResponse, HttpStatus.METHOD_NOT_ALLOWED);
+    }
+
+    /*
+    Exception Handling for Validations to check HttpMessageNotReadable
+     * @param ex HttpMessageNotReadable exception to handle
+	 * @param request the current request
+	 * @return a  ResponseEntity for the response to use
+     */
+
+    @ExceptionHandler(HttpMessageNotReadableException.class)
+    public ResponseEntity<Object> handleHttpMessageNotReadableException(
+            HttpMessageNotReadableException ex, WebRequest request) {
+        ExceptionResponse exceptionResponse =
+                new ExceptionResponse(ex.getMessage());
+        return new ResponseEntity<Object>(exceptionResponse, HttpStatus.BAD_REQUEST);
+    }
+
+    /*
+    Exception Handling using Validations to check MethodArgumentNotValidException
+     Using handleMethodArgumentNotValid
+     to check MethodArgumentNotValidException
+     to validate @NotNull,@Pattern,@Digits,@DecimalMin,@Min and @Max
+	 * @param  MethodArgumentNotValidException ex the exception to handle
+	 * @param request the current request
+	 * @return a  ResponseEntity for the response to use
+     */
+
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<Object> handleMethodArgumentNotValidException(
+            MethodArgumentNotValidException ex, WebRequest request) {
+        ErrorModelResponse error = new ErrorModelResponse();
+        for (FieldError fieldError : ex.getBindingResult().getFieldErrors()) {
```

Comment:
change the code here to -> ex.hasFieldErrors()?ex.getFieldErrors().stream().map(error -> error.isBindingFailure() ?
"Invalid value "+ error.getRejectedValue() " for field" + error.getField()
: 
"The field " + fieldError.getField() + "  has  " + fieldError.getDefaultMessage()))